### DISCUSSION TITLE
How to run a scheduled job as a one off

### DISCUSSION THREAD

**Discussion by securitypedant** [agronholm/apscheduler#709] [General]
Hi

I have a Flask application that uses apscheduler to run an internet speed test every 3 hours. I also want in my web UX to allow the user to run the same job immediately. But I only want one instance of the speed test to be running at any one time.

How can I...

1. Determine if a specific job is running or not.
2. Start a job immediately, ignoring the interval schedule it's on.

**Comment by agronholm** [comment]
There's sadly no way to do either with APScheduler v3.x. APScheduler 4.x currently supports item 2 with its `add_job()` method (which is not at all like `add_job()` on 3.x). I intend for 4.x to have some way to do item 1 too.

For your use case, you might want to consider distributed locking. How you would do it depends on what external services you have available in your deployment. Redis and PostgreSQL are good examples.
